faunadb-scala/src/main/scala/faunadb/values/time/package.scala

```diff
@@ -0,0 +1,89 @@
+package faunadb.values
+
+import org.joda.time.Instant
+import org.joda.time.format.ISODateTimeFormat
+import scala.Ordered._
+
+package time {
+
+  /**
+    * Represents an instance of [[org.joda.time.Instant]] with nanosecond precision.
+    * It calculates the overflow from nano to milliseconds and adds to the initial instant.
+    */
+  class HighPrecisionTime private(initialInstant: Instant, nanosToAdd: Long) {
+
+    /** [[org.joda.time.Instant]] representation. Truncates micro and nanoseconds. */
+    val instant = initialInstant.plus(nanosToAdd / 1000000)
+
+    /** Nanoseconds added to the initial time */
+    val nanos = nanosToAdd % 1000000
```

Comment:
it's confusing to have nanos and micros return some fraction, and millis return the timestamp.